agent/metrics_test.go

```diff
@@ -205,6 +205,29 @@ func TestAgent_OneTwelveRPCMetrics(t *testing.T) {
 	})
 }
 
+func TestHTTPHandlers_AgentMetrics_LeaderShipMetrics(t *testing.T) {
+	skipIfShortTesting(t)
+	// This test cannot use t.Parallel() since we modify global state, ie the global metrics instance
+
+	t.Run("check that we can still turn on consul.http metrics", func(t *testing.T) {
+		hcl := `
+		telemetry = {
+			prometheus_retention_time = "5s",
+			disable_compat_1.9 = false
+			metrics_prefix = "agent_http_2"
+		}
+		`
+
+		a := StartTestAgent(t, TestAgent{HCL: hcl})
+		defer a.Shutdown()
+
+		respRec := httptest.NewRecorder()
+		recordPromMetrics(t, a, respRec)
+
+		assertMetricExists(t, respRec, "is_leader")
+	})
```

Comment:
could we also try to start a follower agent and assert that :

```
assertMetricExistsWithValue(t, respRec, "is_leader", "0")
```

Reply:
@acpana , thanks for the suggestions. To get the correct measurement, I think we need to wait until the metrics is updated. So could you kindly point me to some place where I can update the `MetricsReportingInterval` of the TestAgent? Thanks.

Reply:
it seems like you plumbed it thru 💯